src/main/java/io/curity/identityserver/plugin/awssns/client/SnsClientManagedObject.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 Curity AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.curity.identityserver.plugin.awssns.client;
+
+import io.curity.identityserver.plugin.awssns.config.AwsSnsSmsConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import se.curity.identityserver.sdk.plugin.ManagedObject;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.InstanceProfileCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sts.StsClient;
+import software.amazon.awssdk.services.sts.model.AssumeRoleRequest;
+import software.amazon.awssdk.services.sts.model.AssumeRoleResponse;
+
+public class SnsClientManagedObject extends ManagedObject<AwsSnsSmsConfig>
```

Comment:
Implement the `close` method here, you will be able to close the `SnsClient` object when configuration changes.

Reply:
My bad .., forgot to add that. added now.